### ISSUE TITLE
svelte-check --compiler-warnings ignored when tsconfig.json has compilerOptions

### ISSUE THREAD

**Issue by jnikula** [sveltejs/language-tools#2588]
### Describe the bug

Upgraded to svelte 5. Started getting:

```
tsconfig.json:7:3
Error: 'forceConsistentCasingInFileNames' should be set inside the 'compilerOptions' object of the config json file 
```

Tried to move it to compilerOptions, but simply adding compilerOptions in `tsconfig.json` apparently makes `svelte-check` ignore the `--compiler-warnings` command-line option set in `package.json`. I get a plethora of warnings that should've been ignored.

### Reproduction

Add `--compiler-warnings` option to `svelte-check` in `package.json` to ignore some warning.

Hit that warning.

Add `compilerOptions` object in `tsconfig.json`. The object doesn't have to have anything meaningful, simply its presence will make `svelte-check` ignore the `--compiler-warnings` option.

### Expected behaviour

I'd expect command-line options to amend whatever is in the config file, not to be ignored. The minimum would be a warning about command-line options being incompatible with configuration.

### System Info

- OS: Linux
- IDE: Emacs, command-line

### Which package is the issue about?

`svelte-check`

**Comment by jnikula** [comment]
I tried adding `warningFilter` in `compilerOptions` in `svelte.config.js`, and that can be used to replace the `--compiler-warnings` command-line option. However, that too gets ignored when I set `compilerOptions` in `tsconfig.json`.

I'm pretty much lost on how all these things should be working together. Bottom line, this is blocking my Svelte 5 migration.

**Comment by dummdidumm** [comment]
Please provide a reproduction repository. I have no idea what the warnings are that are supposed to now show up and what the code is that makes them appear.

**Comment by jnikula** [comment]
The repo is https://github.com/groovescore/snooker-scoreboard and the reproduction branches there are:

* [svelte-5](https://github.com/groovescore/snooker-scoreboard/tree/svelte-5) After migration, `npm run check` started complaining `Error: 'forceConsistentCasingInFileNames' should be set inside the 'compilerOptions' object of the config json file`
* [svelte-5-compiler-options](https://github.com/groovescore/snooker-scoreboard/tree/svelte-5-compiler-options) This is `svelte-5` branch plus a simple attempt at moving `forceConsistentCasingInFileNames` inside `compilerOptions`. Leads to `svelte-check --compiler-warnings` being ignored.
* [svelte-5-warning-filter](https://github.com/groovescore/snooker-scoreboard/tree/svelte-5-warning-filter) This is `svelte-5` branch plus first switching from `--compiler-warnings` to `warningFilter`, and then the same simple attempt as above. Leads to the `warningFilter` being ignored.

I just don't know how to both keep the warning ignores/filters I have *and* fix the new warning introduced during the migration.

**Comment by jasonlyu123** [comment]
It's not that the `--compiler-warnings` is ignored when there is `compilerOptions` in your tsconfig. It's because you encounter two breaking changes from svelte-check and svelte. In svelte-check v4, errors in tsconfig.json will prevent further checks. And in svelte 5, the [warning is renamed](https://svelte.dev/docs/svelte/v5-migration-guide#Other-breaking-changes-Error-warning-codes-have-been-renamed). 

Maybe we could map the `--compiler-warnings` to its new name or issue a warning when the flag is passed with the old naming schema, i.e., `-` instead of `_`. But because of how the warning filtering works, we can't map the code to the new code for you. You have to use the new names.

**Comment by jnikula** [comment]
@jasonlyu123 Many thanks! That cleared up the issue. I had glanced through the migration guide, but obviously missed the relevance of the warning renames. PEBKAC. I used the migration script, and I guess in a perfect world it could've renamed the warning filters too!

I should also note that my `warningFilter` was broken and used `switch` on `warning` instead of `warning.code`. Other than that, I guess I'll switch from `--compiler-warnings` to `warningFilter` because it keeps `npm run build` and `npm run dev` output clean too.

Thanks for the quick support!
